testnet/stacks-node/src/config.rs

```diff
@@ -1209,20 +920,56 @@ impl Config {
             node.set_deny_nodes(deny_nodes, burnchain.chain_id, burnchain.peer_version);
         }
 
+        // Validate the node config
+        if is_mainnet {
+            if node.use_test_genesis_chainstate == Some(true) {
+                return Err(format!(
+                    "Attempted to run mainnet node with `use_test_genesis_chainstate`"
+                ));
+            }
+        } else if node.require_affirmed_anchor_blocks {
+            // testnet requires that we use the 2.05 rules for anchor block affirmations,
+            // because reward cycle 360 (and possibly future ones) has a different anchor
+            // block choice in 2.05 rules than in 2.1 rules.
+            debug!("Set `require_affirmed_anchor_blocks` to `false` for non-mainnet config");
+            node.require_affirmed_anchor_blocks = false;
+        }
+
+        let miners_contract_id = boot_code_id(MINERS_NAME, is_mainnet);
```

Comment:
I assumed it was better to forcibly add subscription to the stacker db miners contract rather than to error out. But let me know if this is not the case.